orangecontrib/text/widgets/tests/test_owannotator.py

```diff
@@ -197,6 +197,15 @@ def test_attr_label_metas(self):
         simulate.combobox_activate_item(self.widget.controls.attr_label,
                                         self.corpus.domain[-1].name)
 
+    def test_attr_heuristics(self):
+        widget = self.create_widget(OWAnnotator)
+        c = preprocess(Corpus.from_file("slo-opinion-corpus.tab"))
+        self.send_signal(widget.Inputs.corpus, c[:20])
+        self.wait_until_finished()
+        controls = widget.controls
+        self.assertEqual(controls.attr_x, "t-SNE-x")
```

Comment:
`controls.attr_x` is a `QCombo` and not a string.
string is a corresponing property `widget.attr_x.name`.
So you can either do:
`self.assertEqual(controls.attr_x.currentText(), "t-SNE-x")`
or
`self.assertEqual(widget.attr_x.name, "t-SNE-x")`.

Another thing, your data has PCA instead of tSNE.